### DISCUSSION TITLE
DB::dd()->... like with Http::dd()->...

### DISCUSSION THREAD

**Discussion by shaedrich** [laravel/framework#38037] [Ideas] [answered]
It's be nice to have a `dd()` method for queries as well like in `Http::dd()->` since it's often needed to have the query (especially with bindings):

**Comment by backstageel** [comment, marked as answer]
But there is already this method on DB Facade:

`\DB::table('product_bags')->whereNotNull('bag_id')->limit(10)->dd();`

**Reply by shaedrich** [reply]
Oh, thanks. How could I have missed that? ^^ 🤦

**Reply by mbroersen** [reply]
If you would like to get more out of query dumps I created a package called
[socialblue/laravel-query-adviser](https://github.com/socialblue/laravel-query-adviser)

which returns more information about your query
```php
[
  "toSql" => "select * from `users` where `name` = ? and `users`.`deleted_at` is null"
  "bindings" => [
    0 => "mb"
  ]
  "query" => "select * from `users` where `name` = 'mb' and `users`.`deleted_at` is null"
  "optimizeQuery" => []
]
```

Instead of the default dd
```
"select * from `users` where `users` = ? and `users`.`deleted_at` is null"
[
  0 => "mb"
]

```
The package has more options, but this is one of them
